src/internal.c

```diff
@@ -1975,20 +1982,30 @@ int ChannelPutData(WOLFSSH_CHANNEL* channel, byte* data, word32 dataSz)
         return WS_FATAL_ERROR;
     }
 
-    if (inBuf->length < inBuf->bufferSz &&
-        inBuf->length + dataSz <= inBuf->bufferSz) {
+    if (!(inBuf->length < inBuf->bufferSz &&
+        inBuf->length + dataSz <= inBuf->bufferSz))
+    {
+        /* During rekeying we must buffer either
+         * incoming or outgoing channel data until rekeying is complete.
+         * This means growing the buffer, within reason. */
+        if (channel->ssh->isKeying && dataSz + inBuf->length - inBuf->idx < 1024 *1024 *100) {
```

Comment:
if the upper bounds here of (1024*1024*100) is arbitrary then it should be a definable macro, or maybe related to the max window size